docs/developers/config_tin.py

```diff
@@ -0,0 +1,76 @@
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Optional
+from enum import Enum
+
+
+class ArmoryMode(Enum):
+    HOST = "host"
+    DOCKER = "docker"
+    OFFLINE = "offline"
+
+
+@dataclass
+class ArmoryCredentials:
+    github_token: str
+    s3_id: str
+    s3_key: str
+
+
+@dataclass
+class ArmoryPaths:
+    output_dir: Path
+    local_git_dir: Path
+
+
+def paths_by_mode(mode: str) -> ArmoryPaths:
+    if mode == "docker":
+        return ArmoryPaths(output_dir=Path("/output"), local_git_dir=Path("/git"))
+    elif mode == "host":
+        return ArmoryPaths(
+            output_dir=Path("/tmp/armory"), local_git_dir=Path("/tmp/armory-local")
+        )
+    else:
+        raise ValueError(f"Unknown mode: {mode}")
+
+
+@dataclass
+class ArmoryFlags:
+    interactive: bool = False
+    jupyter: bool = False
+    host_port: Optional[str] = None
+    check_run: bool = False
+    num_eval_batches: Optional[int] = None
+    skip_benign: bool = False
+    skip_attack: bool = False
+    skip_misclassified: bool = False
+    validate_config: bool = False
+
+
+@dataclass
+class ConfigurationTin:
+    mode: str
```

Comment:
using `pydantic` we would just set defaults here like `mode: str = 'native'`.  Can we do that with dataclasses (probably so).  Also, if we go with dataclass, can we do validation on construction?